flink-examples/flink-examples-batch/pom.xml

```diff
@@ -48,25 +48,6 @@ under the License.
 	
 	<build>
 		<plugins>
-			<!-- Fail compilation on deprecation warnings to prevent from showing users outdated examples. -->
-			<plugin>
-				<groupId>org.apache.maven.plugins</groupId>
-				<artifactId>maven-compiler-plugin</artifactId>
-				<executions>
-					<execution>
-						<id>compile</id>
-						<phase>process-sources</phase>
-						<goals>
-							<goal>compile</goal>
-						</goals>
-						<configuration>
-							<compilerArgument>-Xlint:deprecation</compilerArgument>
-							<failOnWarning>true</failOnWarning>
-						</configuration>
-					</execution>
-				</executions>
-			</plugin>
-			
```

Comment:
I wonder if we should simply remove this module. The original purpose is to not showing user outdated examples. With DataSet API being deprecated, we don't want any newcomers to use it.

Reply:
And we probably should also remove the DataSet related documentations.

Reply:
Fixed.

Reply:
Thanks @xintongsong , @WencongLiu ,
I make a PR for tracking: https://github.com/apache/flink/pull/23135
will request a review after going through it with myself.